lib/floe/workflow/choice_rule/data.rb

```diff
@@ -40,52 +61,91 @@ def true?(context, input)
             lhs >= rhs
           when "StringMatches"
             lhs.match?(Regexp.escape(rhs).gsub('\*', '.*?'))
-          else
-            raise Floe::InvalidWorkflowError, "Invalid choice [#{compare_key}]"
           end
         end
 
         private
 
-        def validate!(value)
-          raise "No such variable [#{variable}]" if value.nil? && !%w[IsNull IsPresent].include?(compare_key)
+        def presence_check(context, input)
+          ret_value = compare_value(context, input)
+          # don't need the value, just need to see if the path finds the value
+          variable_value(context, input)
+
+          # path found the variable_value, (so if they said true, return true)
+          ret_value
+        rescue Floe::ExecutionError
```

Comment:
You're rescuing ExecutionError here but in raise_runtime_error! https://github.com/ManageIQ/floe/pull/189/files#diff-d68e675c8b56044da38fd89268ac291a59a7475ae4c31d98a09e76e041092fe3R27 you raise a PathError is that right?

Reply:
Oh nevermind it looks like you catch PathError then re-raise ExceutionError

Reply:
not sure a better way to deal with the whole path errors. I want good error messaging saying where the path was defined and the problem. For both runtime and parse time issues.